File: 0605-can-place-flowers/0605-can-place-flowers.js
/**
 * @param {number[]} flowerbed
 * @param {number} n
 * @return {boolean}
 */
var canPlaceFlowers = function(flowerbed, n) {
  
    //iterate through all the flowerbed and make sure precon is fulfilled
    for (let i = 0; i < flowerbed.length && n > 0; i++){
      //check the previous, current, and next flowerbed whtr is empty
      if(!flowerbed[i-1] && !flowerbed[i] && !flowerbed[i+1]){
        //if its empty make current flowerbed is 1
        flowerbed[i] = 1;
        //decrease the number of n
        n--;
      }
    }
  //if n = 0, successfully planted all n
  return n===0;
};

File: 0605-can-place-flowers/0605-can-place-flowers.dart
class Solution {
  bool canPlaceFlowers(List<int> flowerbed, int n) {
    int canBe = 0;

    for(int i = 0; i < flowerbed.length; i++){
        if(flowerbed[i] == 0){
            bool leftEmpty = (i == 0) || (flowerbed[i -1] == 0);
            bool rightEmpty = (i == flowerbed.length - 1) || (flowerbed[i + 1] == 0);

            if(leftEmpty && rightEmpty){
                flowerbed[i] = 1;
                canBe++;
                i++;
            }
        }

        if(canBe >= n){
            return true;
        }
    }
    return canBe >= n;
  }
}